### ISSUE TITLE
[Kafka Enginee] Json format support.

### ISSUE THREAD

**Issue by kainever7** [ClickHouse/ClickHouse#8160]
Does it support the following json format?

```
{
    "header":{
        "protocol_version":"1",
        "appid":"APP",
		....
    },
    "event":{
        "eventtime":"2019-06-24 18:56:01",
        "type":"event",
        "event":"appUsage",
        "properties":{
            "udid":"863525535942492"
			....
        }
    }
}
```

**Comment by filimonov** [comment]
Currently only JSONEachRow is supported as input format. 
After fixing https://github.com/ClickHouse/ClickHouse/issues/8056 it can be done via custom template.

**Comment by HungUnicorn** [comment]
@filimonov I receive `Cannot parse JSON string: expected opening quote: while reading the key of properties`. any suggestion?

**Comment by alexeydubinin** [comment]
+1. We at hh.ru want to use clickhouse with our kafka based event system and can't undetstand how to extract info from complex jsons. I think there must be solution to map kafka json event to string and extract required info with materialized view and json processing functions (like JSONExtractUInt)

**Comment by filimonov** [comment]
It will be fixed here: #10163
